src/main/java/dataLoader/csvDataLoader.java

```diff
@@ -0,0 +1,33 @@
+package dataLoader;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+
+/**
+ * Simple class that parses data set from a csv file
+ * without knowing the headers of the file
+ * */
+
+public class csvDataLoader {
+
+    private String location = "/path/to/file.csv";
+    private Reader csvData = new StringReader(location);
```

Comment:
this should be moved into the function. In fact the way you have created this class we can only load from one location. If we want to load a file from another location we need to create another instance of it. I don't think currently we want to go this way....what do you think?

Reply:
check out the "refactored" version of parseFile method.